MediaWikiKit/MediaWikiKit/MWKLocationSearchResult.h

```diff
@@ -2,10 +2,41 @@
 #import "MWKSearchResult.h"
 @import CoreLocation;
 
+/**
+ *  Response object model for search results which have geocoordinates.
+ *
+ *  @warning This object only supports deserialization <b>from</b> JSON, not serialization <b>to</b> JSON.
+ */
 @interface MWKLocationSearchResult : MWKSearchResult<MTLJSONSerializing>
 
+/**
+ *  Location serialized from the first set of coordinates in the response.
+ */
 @property (nonatomic, strong, readonly) CLLocation* location;
 
+/**
+ *  Number of meters between the receiver and the coordinate parameters of the originating search.
+ */
 @property (nonatomic, assign, readonly) CLLocationDistance distanceFromQueryCoordinates;
 
+/**
+ *  @name Automatically Updating Properties
+ *
+ *  These properties are set automatically by @c WMFNearbyViewModel in response to location & heading updates.
+ *
+ *  @see -[WMFNearbyViewModel autoUpdateResultAtIndex:]
+ */
+
+/**
+ *  Distance in meters from the user's current location.
+ */
+@property (nonatomic, assign) CLLocationDistance distanceFromUser;
```

Comment:
Mutability is your enemy…  

Also this is conflating the user position (which is changing) with a model object that represents just a search result (which is static).

`distanceFromUser` and `bearingToLocation` really should be somewhere more appropriate - maybe another model object, or maybe cache these values in the view model. 

The nearby view model makes more sense as that is where user position and search results are actually used to calculate `distanceFromUser` and `bearingToLocation`.